### ISSUE TITLE
No nullable warning when lock'ing a potentially null object

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#34711]
**Version Used**:
3.0.0-beta4-19170-01+1deafee3682da88bf07d1c18521a99f47446cee8

**Repro**:
```C#
#nullable enable
class Program
{
    static void Main()
    {
        object? obj = null;
        lock (obj) { }
    }
}
```

I expected a warning on `lock (obj)`, but none is issued.  In contrast, one is issued for `Monitor.Enter` (this is using a corelib that already has Monitor annotated):
![image](https://user-images.githubusercontent.com/2642209/55447727-e5c52700-5592-11e9-99b4-2a05ffbcd5d0.png)

cc: @cston, @dotnet/nullablefc

**Comment by cston** [comment]
@stephentoub, thanks for reporting. This has been fixed in https://github.com/dotnet/roslyn/pull/34004.

**Comment by stephentoub** [comment]
Thanks, @cston.
